compiler/ilgen/OMRJitBuilderRecorderBinaryFile.cpp

```diff
@@ -0,0 +1,43 @@
+/*******************************************************************************
+ * Copyright (c) 2016, 2017 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at http://eclipse.org/legal/epl-2.0
+ * or the Apache License, Version 2.0 which accompanies this distribution
+ * and is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#include <stdint.h>
+#include <fstream>
+
+#include "ilgen/JitBuilderRecorderBinaryFile.hpp"
+#include "infra/Assert.hpp"
+
+// OMR::JitBuilderRecorderBinaryFile::JitBuilderRecorderBinaryFile(const TR::MethodBuilderRecorder *mb, const char *fileName)
+OMR::JitBuilderRecorderBinaryFile::JitBuilderRecorderBinaryFile(const char *fileName)
+   : TR::JitBuilderRecorderBinaryBuffer(), _file(fileName, std::fstream::out | std::fstream::app)
+   {
+   }
+
+void
+OMR::JitBuilderRecorderBinaryFile::Close()
+   {
+   end();
+   EndStatement();
```

Comment:
Call the superclass' implementation to get `end(); EndStatement();` then do the write and close the file here.